### ISSUE TITLE
我的Maix bit SD卡不识别

### ISSUE THREAD

**Issue by nicolaxu** [sipeed/MaixPy_DOC#34]
我在用bit播放视频的时候，video.open("/sd/badapple.avi")，好像根本打不开文件，会直接报错
Traceback (most recent call last):
  File "<stdin>", line 20, in <module>
OSError: [Errno 22] EINVAL
这是SD不兼容还是其他什么问题，我手上两张三星的SD卡，一个8G的，一个32G的，是对容量有什么要求吗？

**Comment by Neutree** [comment]
容量<=128ＧＢ都测试可用，卡是使用SPI协议,部分卡不兼容,测试过一些卡基本都可以,有几款不知名的卡还有一张三星卡不行,其它的比如京东联名卡 闪迪都可以用(class10)

**Comment by Neutree** [comment]
可以跑一下maixduino的SD卡例程,看是不是能支持

**Comment by bluejazzCHN** [comment]
where is SD demo？

**Comment by bluejazzCHN** [comment]
https://github.com/sipeed/MaixPy_scripts/blob/master/multimedia/image/demo_save_picture.py. it takes pic, save SD and load pic to display.

**Comment by SaxonRah** [comment]
### TL;DR 
Change the python string to a raw string and change the forward slashes to backslashes.
> `我在用bit播放视频的时候，video.open("/sd/badapple.avi")，好像根本打不开文件，会直接报错`
read
>`我在用bit播放视频的时候，video.open(r"\sd\badapple.avi")，好像根本打不开文件，会直接报错`

### Actual Answer:
> Forward slashes '/' must be escaped if used in a path, if not you will result in an error 22 EINVAL. Which is actually a linux error code for invalid argument. This makes sense, since Sipeed's MaixPy is running on a FreeRTOS based operating system. uPy(micropython) cannot find the file because it is not at "/sd/badapple.avi" it's at really at "\sd\badapple.avi".

> If you do this, you'll have '\/' spread around your code and the file system. Not desirable.
I suggest that you change the forward slashes to backward slashes. / -> \ because of the undesired look of the string argument.
Unfortunately, this will lead to also more unexpected consequences with uPy since \ is an escape character in strings in python. take \n and \r for example. You can see how this might be bad.
`我在用bit播放视频的时候，video.open("\/sd\/badapple.avi")，好像根本打不开文件，会直接报错`

> So the real solution is converting the string into a python 'raw string'. As well as changing the forward slashes to backwards slashes. Raw string and python strings are exactly the same in memory with escaped characters. However python will not try implicitly escaping the string and leaves it as is. 

We finally get the solution.
`我在用bit播放视频的时候，video.open(r"\sd\badapple.avi")，好像根本打不开文件，会直接报错`

**Comment by Neutree** [comment]
that's not  the string's error, it because can not find file, actually it should raise ENOENT but not EINVAL,  update at commit 14877709e2527c8ef24c9e691d0c1a2205db6870
